src/module_wrap.cc

```diff
@@ -579,31 +590,43 @@ static MaybeLocal<Promise> ImportModuleDynamically(
   }
 
   Local<Value> object;
-
-  int type = options->Get(context, HostDefinedOptions::kType)
-                 .As<Number>()
-                 ->Int32Value(context)
-                 .ToChecked();
-  uint32_t id = options->Get(context, HostDefinedOptions::kID)
-                    .As<Number>()
-                    ->Uint32Value(context)
-                    .ToChecked();
-  if (type == ScriptType::kScript) {
-    contextify::ContextifyScript* wrap = env->id_to_script_map.find(id)->second;
-    object = wrap->object();
-  } else if (type == ScriptType::kModule) {
-    ModuleWrap* wrap = ModuleWrap::GetFromID(env, id);
-    object = wrap->object();
-  } else if (type == ScriptType::kFunction) {
-    auto it = env->id_to_function_map.find(id);
-    CHECK_NE(it, env->id_to_function_map.end());
-    object = it->second->object();
+  if (options->Length() == HostDefinedOptions::kLength) {
```

Comment:
What are we getting as host defined options for import coming from shadow realms? Is it just empty fixed array? I think it would be more natural to tweak this branch as `if (realm->kind() == Realm::Kind::kPrincipal)`..

Reply:
> I think it would be more natural to tweak this branch as `if (realm->kind() == Realm::Kind::kPrincipal)`

Only the import initiated by `ShadowRealm.prototype.importValue` has the empty host-defined option. The referrer script is still present for `import` statement and dynamic import so the host-defined option would not be empty. In that case, this branch applies in ShadowRealm too.